src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12753,23 +12753,32 @@ private ExpressionSyntax ParseObjectOrCollectionInitializerMember(ref bool isObj
         {
             if (this.IsComplexElementInitializer())
             {
+                // { ... }
                 return this.ParseComplexElementInitializer();
             }
             else if (IsDictionaryInitializer())
             {
+                // [...] = { ... }
+                // [...] = ref <expr>
+                // [...] = <expr>
                 isObjectInitializer = true;
                 var initializer = this.ParseDictionaryInitializer();
                 initializer = CheckFeatureAvailability(initializer, MessageID.IDS_FeatureDictionaryInitializer);
                 return initializer;
             }
             else if (this.IsNamedAssignment())
             {
+                // Name = { ... }
+                // Name = ref <expr>
+                // Name =  <expr>
                 isObjectInitializer = true;
                 return this.ParseObjectInitializerNamedAssignment();
             }
             else
             {
-                return this.ParseExpressionCore();
+                // <expr>
+                // ref <expr>
```

Comment:
:memo: This case and the dictionary initializer case above remain errors in binding, but the parser now tolerates `ref` (more graceful failure).